### PR TITLE
GuiConsole: Fix autodock and set it to a small gap

### PR DESCRIPTION
**Pull request by RedDevilus**

### CODE REVIEW

**Comment by lightningterror** [comment]
Was it tested on Windows 8.1, MAC, Linux?

**Comment by RedDevilus** [comment]
Linux: Problem with compiling (not tested)
W8.1: Seems to not move with the main window and sit in a specific spot)
W10: Works perfectly

**Comment by RedDevilus** [comment]
Hotfix results;
Linux: Still needs testing
W8.1: Attaches to main window with no gap
image
W10: Attaches to main window with gap
image

**Comment by lightningterror** [comment]
I'd just leave it as it is honestly.

**Comment by RedDevilus** [comment]
Master autodock for linux (shadows get on top of the console) :
image


PR for linux:
image

**Comment by RedDevilus** [comment]
Reason for this PR is that WX is calculating the topright wrong which gives it an extra 16px gap for no reason in windows but it doesn't do this in Linux. Whereas Linux does have shadows that interfere with the look of the whole program.

**Comment by lightningterror** [comment]
I prefer with gap because of the shadow.

File: pcsx2/gui/MainFrame.cpp [hunk 1]
```diff
@@ -192,7 +192,11 @@ void MainEmuFrame::OnMoveAround(wxMoveEvent& evt)
 		{
 			if (!proglog->IsMaximized())
 			{
-				g_Conf->ProgLogBox.DisplayPosition = GetRect().GetTopRight();
+				// GetSize() is wrong on windows.
+				// GetTopRight() is also wrong on windows since it probably uses GetSize() while it gives the wrong values for autodock.
+				// ConsoleLogger.cpp L713: if( wxWindow* main = GetParent() ) ;always triggers on false and it's corresponding lines never get executed.
+				int xCoord = GetRect().GetTopLeft().x + GetClientSize().GetWidth() + 10; // Adds a 10px gap between the main window and console.
```
Comment (lightningterror) [inline pcsx2/gui/MainFrame.cpp:198]: You can use const here.

**Comment by lightningterror** [comment]
Can someone post a ss how it looks on windows 8.1?

**Comment by RedDevilus** [comment]
8.1 Windows: 
image

**Comment by lightningterror** [comment]
Doesn't look good, main window overlaps with the console.

**Comment by RedDevilus** [comment]
I could change 10 to 15

**Comment by RedDevilus** [comment]
8.1:
image

10:
image

Changed back to original value of 20 but eliminates some issues people could have from wrong calculation for screen position.

**Comment by lightningterror** [comment]
There's still some gap.

**Comment by RedDevilus** [comment]
Making it a bigger gap will make it uglier for w10 just appealing to how w8.1 works.

**Comment by lightningterror** [comment]
I think we should leave this out as it feels more like a hackfix.

**Comment by RedDevilus** [comment]
I could look into make it seperate for 8.1 because of how the gaps work differently. Current code already is a hack on it's own.

**Comment by tellowkrinkle** [comment]
macOS looks like Linux, though it uses less aggressive shadows
Screenshot of PCSX2 with no gap between windows
Screenshot of PCSX2 with gap between windows

**Comment by GovanifY** [comment]
Looks like we like the gap after all

**Pull request closed without merging**